### ISSUE TITLE
Howto deploy a ovf thin provisioned?

### ISSUE THREAD

**Issue by sithglan** [vmware/govmomi#1523]
Hello,
I try to deploy a thin provisioned VM using the following command:

`govc import.ova --options=vra.json -ds=ssd-01.v102 -name=vra-01 /var/www/vra76/VMware-vR-Appliance-7.6.0.317-13027280_OVF10.ova`

In vra.json is the property DiskProvisioning set to thin:

`"DiskProvisioning": "thin",`

However the resulting VM thick provisioned. What did I do wrong?

Cheers,
     Thomas

**Comment by dougm** [comment]
Working as expected here, how are you checking the disk provisioning?

```console
% jq .DiskProvisioning ttylinux.json
"thin"

% govc import.ova -ds datastore1 -options ttylinux.json govc/test/images/ttylinux-pc_i486-16.1.ova
[22-07-19 20:31:11] Uploading ttylinux-pc_i486-16.1-disk1.vmdk... OK

% govc device.info -json -vm ttylinux disk-* | jq '.Devices[].Backing | .FileName, .ThinProvisioned'
"[datastore1] ttylinux/ttylinux.vmdk"
true
```

**Comment by sithglan** [comment]
Hello Doug,

> Working as expected here, how are you checking the disk provisioning?

I checked in the vSphere Client and by comparing 'du -sh' with 'ls -alh'
on the ESX. However I tried what you did and was unable to reproduce. If
it happens again to me, I will make it reproducable and reopen an issue.

''
govc device.info -json -vm vra-14 disk-* | jq '.Devices[].Backing | .FileName, .ThinProvisioned'
"[ssd-14.v102] vra-14/vra-14.vmdk"
true
"[ssd-14.v102] vra-14/vra-14_1.vmdk"
true
"[ssd-14.v102] vra-14/vra-14_2.vmdk"
true
"[ssd-14.v102] vra-14/vra-14_3.vmdk"
true
''

Cheers,
        Thomas

**Comment by dougm** [comment]
On ESX du/ls will show the capacity rather than actual space used iirc.

How did you check using vSphere client?  We could also add this detail to device.info output.

**Comment by sithglan** [comment]
Hello Doug,

> On ESX du/ls will show the capacity rather than actual space used
> iirc.

for me it shows:

[root@node-64:/vmfs/volumes/511f3a5d-fda86f90-61ca-00259097dbae/w2k16-master-64] ls -alh *flat.vmdk
-rw-------    1 root     root       80.0G Jul 22  2018 w2k16-master-64-flat.vmdk
[root@node-64:/vmfs/volumes/511f3a5d-fda86f90-61ca-00259097dbae/w2k16-master-64] du -sh *flat.vmdk
13.0G   w2k16-master-64-flat.vmdk

> How did you check using vSphere client?

I looked at the summary tab of the VM under VM Hardware, Hard Disk 1 and
Type. But since I can't reproduce it, I have no idea what happened. And
a few strange things happened. What I did from memory was:

I deployed 14 vra and 14 iaas (W2k16) servers. They were full
provisioned and for at least one iaas the vmdk was double the size of
the configured logical disk space which did not make any sense for me
whatsoever. The vmdk physical and logical size was 160GB but the hard
disk was configured at 80 GB. Find the scripts that I used to deploy the
VMs attached. Also notice that for the iaas (w2k16) I did not specify a
json file but assumed that because when I generated the json file it
already said 'thin', when I deploy it it is already thin. Find the
scripts attached.

> We could also add this detail to device.info output.

I'm only getting started with govc but since I'll work more with vSphere
6.7 environments in the future I'll prefer govc over the html5 client.
Thank you for providing it. If it would not be there I would write my
own client in perl.

Cheers,
        Thomas
